README.markdown

```diff
@@ -23,72 +25,53 @@ $ ./scripts/update
+setup, next.
+ 
 #### Django Application Setup ####
 
 The django application includes a management command that handles application configuration
 based on two input files: `settings.py` in the `publicmapping` app in `django/publicmapping`
-and `config.xml`. Below is an explanation of what each of these flags does:
-
-##### Internationalization #####
-
-To compile language files, use the `-l` flag.
-
-##### Static Files #####
-
-To collect static files, use the `-s` flag.
-
-##### Database Views #####
-
-The `-V` flag creates some kind of database views. It doesn't currently work though.
+and `config.xml`.
 
 ##### Data Setup #####
 
-The following flags are all executed as part of `scripts/load_development_data`.
-
-###### `-gX`: Loading Geolevels ######
-
-The `-gX` flag loads geolevel `X` into the database. Where to find those geolevels is determined
-by the `config.xml` file. Each geolevel in the xml has an entry like:
+The first script to run is `./scripts/load_development_data`. It is not fast, and it's currently
+recommended that you run it overnight. We specifically would like to improve performance of the
+data load.
+
+This script will do several things
+
+- Fetch zipped shapefile data for Virginia into a specific location
+- Drop and recreate the `district_builder**
+- Run database migrations: create the relationships that data will be loaded into
+- Load shapes from shapefiles at different levels: create records for the shapes and characteristics
+  in the configured shapefiles
+- Nest the shapes at different levels into each other: calculate the spatial relationships between
+  shapes at different zoom levels
+- Load some template plans: initialize the database with several example plans that users can start
+  drawing with
+- Create database views: create the database objects that GeoServer will use to
+  create tiles of specific subjects
+- Configure GeoServer: create layers and styles that will be served as tiles to the frontend
+
+**IMPORTANT**: after you configure GeoServer, from outside the vm, run `vagrant rsync-back`. If you
+don't yet have that plugin, run `vagrant plugin install vagrant-rsync-back`. If you don't do this,
+your host won't pick up the files that were created in GeoServer configuration, so when you reboot
+your vm, the files won't be present in the guest either, and then they won't be in the GeoServer
+docker container. That will break GeoServer in a confusing and surprising way, and you'll be sad,
+so just make sure you `rsync-back`.
 
-```xml
-<Shapefile path="/data/some_path.shp">
-  <Fields>
-  ...
-  </Fields>
-</Shapefile>
-```
-
-that tells the management command where to look for geounits in this level, and what fields to
-care about in each shapefile record. The lowest numbered geolevel is for the smallest size, so
-if you had only counties and states, for example, counties would be `-g0`, and states would be
-`-g1`.
-
-###### `-nX`: Nesting Geolevels ######
```

Comment:
You don't think this flag-specific documentation is useful (geolevels, templates, geoserver, database views)?

Reply:
I thought it wasn't, since we're bundling all that up in `load_development_data`. I'll bring it back though, maybe with a reorganization that makes it more obvious that you don't _need_ to care about it, but that you can care about it if you want.

Reply:
Okay, sounds good. My understanding was that `load_development_data` was mostly for our use to make our lives easier (VA data is hard-coded). Is that meant to be used by anyone eventually?

Reply:
I don't know, maybe. We need to make configuration easier for people someday, and this does a lot of the processing of the configuration, so I can imagine it being used by the public with a configurable destination for where the zipped data live maybe.